server/src/main/java/com/cloud/api/ApiServlet.java

```diff
@@ -213,7 +213,7 @@ void processRequestInContext(final HttpServletRequest req, final HttpServletResp
                     try {
                         responseString = apiAuthenticator.authenticate(command, params, session, remoteAddress, responseType, auditTrailSb, req, resp);
                         if (session != null && session.getAttribute(ApiConstants.SESSIONKEY) != null) {
-                            resp.addHeader("SET-COOKIE", String.format("%s=%s;HttpOnly", ApiConstants.SESSIONKEY, session.getAttribute(ApiConstants.SESSIONKEY)));
+                            resp.addHeader("SET-COOKIE", String.format("%s=%s;HttpOnly;Path=/client", ApiConstants.SESSIONKEY, session.getAttribute(ApiConstants.SESSIONKEY)));
```

Comment:
We need to check if this is necessary or not

Reply:
Looks like it is necessary, if not present, two sessionkeys are present
If non-standard implementations are the case, wouldn't JSESSIONID cause issues too?
Also exploring saml logins

Reply:
@davidjumani in a deploy trillian env with this PR (or you may use the primate-qa server as well which has this PR) and both old UI and Primate, can you check side-effects when you change the context in /etc/cloudstack/management/server.properties from `/client/` to say (a) `/somenew-path` and (b) `/` and restart management server and try log-in in both legacy UI and Primate (check network an Application/cookies)

Reply:
Nevermind I tested, this will break if path is restricted to /client only; ideally shoud match the webapp context path